lib/postgres.dart

```diff
@@ -149,6 +152,7 @@ abstract class SessionExecutor {
   Future<R> runTx<R>(
     Future<R> Function(Session session) fn, {
     TransactionMode? transactionMode,
+    // SessionSettings? sessionSettings,
```

Comment:
Maybe we could also have a `TransactionSettings` class inheriting from `SessionSettings` to add transaction-specific properties. I didn't fully think this through yet, it just might make the API a bit simpler. That could also include the timeout.

Reply:
Good point. Maybe we can reduce the parameter name into `settings` in various places (e.g. both `Connection` and `Pool` only needs one `settings`.

Reply:
Updated the code (not fully implemented every detail yet). wdyt?

Reply:
That looks like what I had in mind :+1: